pya/backend/PyAudio.py

```diff
@@ -0,0 +1,49 @@
+import pyaudio
+from .base import BackendBase
+
+
+class PyAudioBackend(BackendBase):
+
+    def __init__(self, format=pyaudio.paFloat32):
+        self.pa = pyaudio.PyAudio()
+        self.format = format
+        if format == pyaudio.paInt16:
+            self.dtype = 'int16'
+            self.range = 32767
+        elif format == pyaudio.paFloat32:
+            self.dtype = 'float32'  # for pyaudio.paFloat32
+            self.range = 1.0
+        else:
+            raise AttributeError(f"Aserver: currently unsupported pyaudio format {self.format}")
```

Comment:
should the error be reported as coming from Aserver or better PyAudioBackend?

Reply:
this is related to `format` passed to the backend. `format` -- which can either be `pyaudio.paFloat32` or `pyaudio.paFloat32` -- is a `int` which is processed by pyaudio. The int value however has no semantic connections to whether floats or ints are actually expected for an audio stream. Thus, I consider this a contextual variable which should be handled by the backend. Other backends might support other number schemes or formats which should be transparent to `Aserver`. The error string however is misleading and i'd change it to `"PyAudioBackend: ..."`